tgui/packages/tgui/interfaces/GasCanister/DetonatorTimer.js

```diff
@@ -0,0 +1,35 @@
+import { Box, AnimatedNumber } from '../../components';
+import { formatTime } from '../../format';
+
+export const DetonatorTimer = props => {
+  const {
+    time,
+    warningThreshold = 30,
+    dangerThreshold = 10,
+    explosionMessage = "BO:OM",
+  } = props;
+
+  const timeColor = () => {
```

Comment:
As you're just using once, curious of reason for function here rather than just:
```
let timeColor = 'green';
if (time <= dangerThreshold) {
    timeColor = 'red';
} else if (time <= warningThreshold) {
    timeColor = 'orange';
}
// ...
<Box
    // ...
    color={timeColor}
    // ...

Reply:
Probably glazed over it during cleanup. Updated.